adb/adb_commands.py

```diff
@@ -127,10 +184,32 @@ def Install(self, apk_path, destination_dir='', timeout_ms=None):
     if not destination_dir:
       destination_dir = '/data/local/tmp/'
     basename = os.path.basename(apk_path)
-    destination_path = destination_dir + basename
+    destination_path = os.path.join(destination_dir, basename)
```

Comment:
Concatenating the string often got the path wrong ('/data/local/tmptest' vs '/data/local/tmp/test'). Using os.path.join to ensure it produces a correct path.